sinch/domains/verification/models/responses.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from sinch.core.models.base_model import SinchBaseModel
+
+
+@dataclass
+class StartVerificationResponse(SinchBaseModel):
+    id: str
+    method: str
+    sms: dict
+    _links: list
+
+
+@dataclass
+class VerificationResponse(SinchBaseModel):
```

Comment:
This do not fit with specs (SMS, FlashCall, Phone Call) and may be naming class as `StartVerificationResponse`
Like said above: we should support enums as helpers to define managed values when it come to a set of restricted values status, reason, ....)

Reply:
Ok this class is not for the StartVerification response.
Getting report and getting status request are returning same object (defined as `VerificationResponse` from OAS specs)
So finally `VerificationResponse` is the only one to be returned and not requiring a `ReportVerificationResponse`
About class fields
  - `status` should be an enum 
  - `reason` field is missing (and should be an enum too)
  - `source`, `callComplete` and `callResult` are missing
  - `identity` is not a string

Reply:
I'll keep VerificationResponse as a superclass due to consistency reasons. I already use that nomenclature across all domains.
I've added some missing fields though.

Reply:
Then because of `ReportVerificationResponse` and `VerificationResponse` are identical `ReportVerificationResponse` should at least inherit from `ReportVerificationResponse`
Currently, the defined `ReportVerificationResponse` do not fit with real server response

(see previous comment: `identity` is still defined as a string but is not: it is a complex type)

Reply:
I don't think we have `identity` on Verification response objects at all :) (according to the docs)

Reply:
Here is a example of response from the server. Please check the `identity` is an object and the specification needs to be updated accordingly
```
{
  "id": "018d3154-2205-7ff6-1b00-666c75432593",
  "method": "flashcall",
  "status": "FAIL",
  "reason": "Expired",
  "reference": "test-reference-for-flashCall-verification",
  "price": {
    "verificationPrice": {
      "currencyId": "EUR",
      "amount": 0.0308
    },
    "terminationPrice": {
      "currencyId": "EUR",
      "amount": 0
    }
  },
  "identity": {
    "type": "number",
    "endpoint": "+33612345678"
  },
  "countryId": "FR",
  "verificationTimestamp": "2024-01-22T13:20:57.9492144",
  "callComplete": true,
  "callResult": "ANSWERED"
}
```

Reply:
I think a object rather than a simple `dict` would be beneficial for a user in general and especially for this case where the documentation doesn't contain anything about its structure. WDYT?

Reply:
For now I would like to stay with simple `dict`. I want to use static annotations in the near future for type hints and leave a lot of freedom to the dev.

Reply:
A counter part about dict vs dedicated classes: it will led to not homogeneous SDK usage and be very coupled to underlying API.
So when API is using camel case, kebab case, snake case it will impact the SDK representation 
Even for a same API we could have differences according to context (`callid` in requests vs `callId` in webhooks )
So like sample above: it is not a python friendly snake_case representation